compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/AbstractParserErrorHandler.java

```diff
@@ -35,15 +35,25 @@
     protected final AbstractTokenReader tokenReader;
     protected final BallerinaParserErrorListener errorListener;
     private ArrayDeque<ParserRuleContext> ctxStack = new ArrayDeque<>();
+    private int previousIndex;
+    private int itterCount;
 
     /**
      * Limit for the distance to travel, to determine a successful lookahead.
      */
     protected static final int LOOKAHEAD_LIMIT = 5;
 
+    /**
+     * Limit for the number of times parser tries to recover staying on the same token index.
+     * This will prevent parser going to infinite loops.
+     */
+    private static final int ITTER_LIMIT = 10;
```

Comment:
How about a limit of 5? will it break things?

Reply:
Yes 5 break things. Trying 7